src/components/TopBanner.js

```diff
@@ -0,0 +1,52 @@
+import _ from 'lodash/fp'
+import { div, h } from 'react-hyperscript-helpers'
+import Interactive from 'react-interactive'
+import { icon } from 'src/components/icons'
+import * as Style from 'src/libs/style'
+import * as Utils from 'src/libs/utils'
+import { Component } from 'src/libs/wrapped-components'
+
+
+/**
+ * @param {bool} isVisible
+ * @param {bool} [showX=true]
+ * @param {function} [onDismiss]
+ */
+export default class TopBanner extends Component {
+  componentDidMount() {
+    Utils.waitOneTick().then(() => this.setState({ show: this.props.isVisible }))
+  }
+
+  static getDerivedStateFromProps(props, state) {
+    return _.isUndefined(state.show) ? null : { show: props.isVisible }
+  }
+
+  render() {
+    const { showX = true, children, onDismiss, ...props } = _.omit('isVisible', this.props)
+    const { show } = this.state
```

Comment:
Two-part comment:
1. Do we want this as a separate component, rather than just part of ErrorBanner?
2. Assuming yes, then perhaps we should avoid creating multi-tiered state, and just have this part be stateless.